source/material_model/melt_phipps_morgan.cc

```diff
@@ -0,0 +1,1363 @@
+                   double &new_molar_composition_of_solid,
+                   double &new_molar_composition_of_melt) const
+    {
+      if (temperature == 0)
+        return 0;
+
+      // TODO PM01 make this a loop over the melts instead of having separate Fe and Mg parameters
+      {
+        // after Phipps Morgan, Jason. "Thermodynamics of pressure release melting of a veined plum pudding mantle."
+        // Geochemistry, Geophysics, Geosystems 2.4 (2001).
+        const double P = pressure;                   // pressure in Pa
+        const double T = temperature;                // temperature in K
+        const double R = constants::gas_constant;    // Ideal Gas Constant
+
+        // Free Energy Change Delta_G due to Melting as a function of temperature and pressure
+        // eq 2 of Myhill 2018
+        const double dG_Fe_mantle =  Fe_delta_E - T * Fe_mantle_melting_entropy + P * Fe_mantle_melting_volume
+                                     + 0.5 * P * P * Fe_delta_V_prime_fusion;
+        const double dG_Mg_mantle =  Mg_delta_E - T * Mg_mantle_melting_entropy + P * Mg_mantle_melting_volume
+                                     + 0.5 * P * P * Mg_delta_V_prime_fusion;
+
+        // c0 and c1 of eq 14 & 15 of Dannberg ea and part of A10 & A11 of Phipps Morgan
+        const double p_Fe_mantle = std::exp(dG_Fe_mantle/(Fe_number_of_moles*R*T));
+        const double p_Mg_mantle = std::exp(dG_Mg_mantle/(Mg_number_of_moles*R*T));
+
+        // Mole composition of Fe in the liquid (eq 12)
+        // A12 in PM01, although
+        // A12 computes X_Fe_l = (1-p_Mg)/(p_Fe-p_Mg)
+        const double Xls = 1. - ((1. - p_Fe_mantle) / (p_Mg_mantle - p_Fe_mantle));
+
+        // Melting occurs when delta G = 0
+        const double T_Fe_mantle = (Fe_delta_E + P * Fe_mantle_melting_volume + 0.5 * P * P * Fe_delta_V_prime_fusion) / Fe_mantle_melting_entropy;
+        const double T_Mg_mantle = (Mg_delta_E + P * Mg_mantle_melting_volume + 0.5 * P * P * Mg_delta_V_prime_fusion) / Mg_mantle_melting_entropy;
+        std::cout << "T, P, New T_Melt " << T << ", " << P << ", " << T_Fe_mantle << ", " << T_Mg_mantle << std::endl;
+
+        // TODO PM01 rm
+        {
+        const double old_dG_Fe_mantle = (Fe_mantle_melting_temperature - T) * Fe_mantle_melting_entropy
+                                    + (P - melting_reference_pressure) * Fe_mantle_melting_volume;
+        const double old_dG_Mg_mantle = (Mg_mantle_melting_temperature - T) * Mg_mantle_melting_entropy
+                                    + (P - melting_reference_pressure) * Mg_mantle_melting_volume;
+        const double old_T_Fe_mantle = old_dG_Fe_mantle / Fe_mantle_melting_entropy;
+        const double old_T_Mg_mantle = old_dG_Mg_mantle / Mg_mantle_melting_entropy;
+        std::cout << "Old T_Melt Fe, Mg: " << old_T_Fe_mantle << ", " << old_T_Mg_mantle << std::endl;
+        } 
+
+        double melt_molar_fraction = 0.;
+
+        // Q! why should the melting T be negative?
```

Comment:
See above. `T_Fe_mantle = T_melting_Fe - T` and `T_Mg_mantle = T_melting_Mg - T`, so a negative value implies we are above the melting temperature of that endmember.